# PLD ships dyninst libraries in %{_libdir} and renames libcommon to libdyncommon
--- systemtap-5.6/configure.ac	2026-09-18 22:07:22.525408503 +0200
+++ systemtap-5.6/configure.ac	2026-09-18 22:07:22.794409838 +0200
@@ -687,7 +687,7 @@
      # Workaround for Dyninst requiring DYNINST_TBB_HAS_VERSION_H on newer TBB versions
      AC_CHECK_HEADERS([tbb/version.h], [DYNINST_CXXFLAGS="$DYNINST_CXXFLAGS -DDYNINST_TBB_HAS_VERSION_H"])
      CPPFLAGS="$CPPFLAGS $DYNINST_CXXFLAGS"
-     for location in /usr/lib64/dyninst /usr/lib/dyninst ${libdir}/dyninst; do
+     for location in ${libdir} /usr/lib64/dyninst /usr/lib/dyninst ${libdir}/dyninst; do
          LDFLAGS="-L${location}"
          LIBS="-lparseAPI ${BOOST_SYSTEM_LIB}"
 	   AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <dyninst/CodeSource.h>]],
@@ -710,7 +710,7 @@
        save_LIBS="$LIBS"
        AC_CHECK_LIB(tbb, TBB_runtime_interface_version, [tbb_LIBS=-ltbb],[tbb_LIBS=])       
        LIBS="$save_LIBS"
-       DYNINST_LIBS="-lparseAPI -lsymtabAPI -linstructionAPI $tbb_LIBS -lcommon"
+       DYNINST_LIBS="-lparseAPI -lsymtabAPI -linstructionAPI $tbb_LIBS -ldyncommon"
      fi
      ;;
 *)  # Use paths in the user-specified prefix
--- systemtap-5.6/stapdyn/Makefile.am	2026-09-18 22:07:22.638731870 +0200
+++ systemtap-5.6/stapdyn/Makefile.am	2026-09-18 22:07:22.797801177 +0200
@@ -40,7 +40,7 @@
 				  dynutil.cxx ../staputil.cxx
 stapdyn_CXXFLAGS = $(AM_CXXFLAGS) @DYNINST_CXXFLAGS@ $(debuginfod_CFLAGS) @WNO_TEMPLATE_BODY@
 stapdyn_LDFLAGS = $(AM_LDFLAGS) @DYNINST_LDFLAGS@ $(debuginfod_LDFLAGS)
-stapdyn_LDADD = -ldl -ldyninstAPI -lsymtabAPI -lpcontrol -lstackwalk -lcommon -ldw -lelf -lpthread $(debuginfod_LIBS)
+stapdyn_LDADD = -ldl -ldyninstAPI -lsymtabAPI -lpcontrol -lstackwalk -ldyncommon -ldw -lelf -lpthread $(debuginfod_LIBS)
 stapdyn_CFLAGS = $(debuginfod_CFLAGS)
 
 dynsdt_SOURCES = dynsdt.cxx dynutil.cxx ../staputil.cxx
